### DISCUSSION TITLE
Is it possible to autocomplete specific variable type by sign?

### DISCUSSION THREAD

**Discussion by Demianeen** [LazyVim/LazyVim#2946] [Q&A] [answered]
Let's say I have a snippet and keyword, or function and variable. Some lazyvim autocompletion like snippet (with ~ suffix, e.g. def~) or variable (with = suffix, e.g. foo=). Can I set the cmp to use those signs for autocompletion? You can see what I mean in the video:

https://github.com/hrsh7th/nvim-cmp/assets/51330172/d0585a6e-c777-4c2f-8dca-2a804de2d6ff

Do you think it is possible to do something like that without rewriting every comparison function for nvim cmp?

**Comment by dpetka2001** [comment, marked as answer]
I'm not aware if you can do this, but I know you can call a specific source for completion with something like
```lua
  {
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      local cmp = require("cmp")

      opts.mapping = vim.tbl_extend("force", opts.mapping, {
        ["<C-l>"] = cmp.mapping.complete({
          config = {
            sources = vim.tbl_filter(function(source)
              return source.name == "luasnip"
            end, opts.sources),
          },
        }),
      })
    end,
  },
```
That way you can only complete snippets for example. Unfortunately, this doesn't solve your problem about variables, but I think you could do something about it by using `entry_filter`. See [here](https://github.com/hrsh7th/nvim-cmp/blob/ce16de5665c766f39c271705b17fff06f7bcb84f/doc/cmp.txt#L654-L675) for an example.
